app/com/arpnetworking/metrics/portal/scheduling/impl/PeriodicSchedule.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2018 Dropbox, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.portal.scheduling.impl;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import models.internal.scheduling.BaseSchedule;
+import net.sf.oval.constraint.NotNull;
+import net.sf.oval.constraint.ValidateWithMethod;
+
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.Optional;
+
+/**
+ * Schedule for a job that repeats periodically.
+ *
+ * @author Spencer Pearson (spencerpearson at dropbox dot com)
+ */
+public final class PeriodicSchedule extends BaseSchedule {
+
+    /**
+     *
+     */
+    public enum Period {
+        /** */
+        HOUR(Duration.ofHours(1)),
+        /** */
+        DAY(Duration.ofDays(1));
+
+        private Duration _duration;
+        Period(final Duration duration) {
+            _duration = duration;
+        }
+
+        public Duration getDuration() {
+            return _duration;
+        }
+
+        /**
+         * Rounds a time backwards to the nearest natural boundary of this period (e.g. __:00 for HOUR, midnight for DAY).
+         *
+         * @param time The time to round.
+         * @return The rounded time, guaranteed to be on a {period}-boundary in {@code time}'s time zone.
+         */
+        public ZonedDateTime floor(final ZonedDateTime time) {
+            final ZonedDateTime t0 = ZonedDateTime.parse("1970-01-01T00:00:00Z");
```

Comment:
I don't believe that this will not work if the target time is in a timezone that is off from Zulu by a non denominational amount. E.g. your timezone is off by X hours and 30 minutes and you want an hourly report. 

This at least warrants a test. Try `Canada/Newfoundland`; e.g. https://en.wikipedia.org/wiki/Newfoundland_Time_Zone